vlm/src/main/scala/geotrellis/contrib/vlm/gdal/VSIPath.scala

```diff
@@ -0,0 +1,150 @@
+package geotrellis.contrib.vlm.gdal
+
+
+import java.net.URI
+
+
+object Schemes {
+  final val FTP = "ftp"
+  final val HTTP = "http"
+  final val HTTPS = "https"
+
+  final val TAR = "tar"
+  final val ZIP = "zip"
+  final val GZIP = "gzip"
+
+  final val FILE = "file"
+
+  final val S3 = "s3"
+
+  final val GS = "gs"
+
+  final val WASB = "wasb"
+  final val WASBS = "wasbs"
+
+  final val HDFS = "hdfs"
+
+  final val FILE_TYPE_TO_SCHEME =
+    Map(
+      TAR -> TAR,
+      "tgz" -> TAR,
+      ZIP -> ZIP,
+      "kmz" -> ZIP,
+      "ods" -> ZIP,
+      "xlsx" -> ZIP,
+      GZIP -> GZIP
+    )
+}
+
+
+object VSIPath {
```

Comment:
Hm, do we want to have `VSIPath` as a function (the current implementation) or we want to have some class (similar to `URI`)? (It's just a question, and both options are good to me)

Reply:
I think both would work. It might be worth it to make it a class if we'd like to do more with the paths.